tokio-postgres/src/query.rs

```diff
@@ -59,7 +69,17 @@ where
     I: IntoIterator<Item = &'a dyn ToSql>,
     I::IntoIter: ExactSizeIterator,
 {
-    let buf = encode(client, &statement, params)?;
+    let buf = if log::log_enabled!(log::Level::Debug) {
+        let params = params.into_iter().collect::<Vec<_>>();
+        log::debug!(
+            "executing statement {} with parameters: {:?}",
+            statement.name(),
+            params,
+        );
+        encode(client, &statement, params)?
```

Comment:
Can you run the params through [`slice_iter`](https://github.com/sfackler/rust-postgres/blob/master/tokio-postgres/src/lib.rs#L229) here? That should minimize extra instantiations of encode.

Reply:
Nevermind, will require a bit more work to make that happen. I'll deal with it on my end.